packages/builder/src/pages/builder/portal/settings/ai/ConfigModal.svelte

```diff
@@ -1,92 +1,66 @@
-<script>
-  import { ModalContent, Label, Input, Select, Toggle } from "@budibase/bbui"
-  import { ConfigMap, Providers } from "./constants"
+<script lang="ts">
+  import { ModalContent, Label, Input, Select } from "@budibase/bbui"
+  import { ConfigMap, Models } from "./constants"
+  import type { ProviderConfig } from "@budibase/types"
 
-  export let config = {
-    active: false,
-    isDefault: false,
-  }
+  export let config: ProviderConfig
+  export let updateHandler: (_config: ProviderConfig) => void
+  export let enableHandler: (_config: ProviderConfig) => void
+  export let disableHandler: (_config: ProviderConfig) => void
 
-  export let saveHandler
-  export let deleteHandler
+  let validation: boolean
+  let edited: boolean = false
 
-  let validation
+  $: isEnabled = config.active && config.isDefault
 
   $: {
     const { provider, defaultModel, name, apiKey } = config
-    validation = provider && defaultModel && name && apiKey
+    validation = Boolean(provider && name && defaultModel && apiKey)
```

Comment:
`apiKey` is always unset for BudibaseAI, does that make this validation incorrect?

Reply:
ConfigModal is never triggered for BudibaseAI, so it's not relevant.